From e88861055d3a032bec44369a109417c74f5ad6ab Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankit-aj.jain@broadcom.com>
Date: Thu, 9 Jan 2025 08:31:07 +0000
Subject: [PATCH] Revert "crypto: api - Add crypto_tfm_get"

This reverts commit e470d423b0f6ffd3977e74d1d1eef70cc58f18dd.

commit e470d42 (crypto: api - Add crypto_tfm_get) introduced an api
which adds a new member in struct crypto_tfm, causing build issues
with fips canister, since, crypto_tfm_get() api is not being used
in v6.1.x so reverting upstream commit.

Signed-off-by: Ankit Jain <ankit-aj.jain@broadcom.com>
---
 crypto/api.c           | 4 ----
 crypto/internal.h      | 6 ------
 include/linux/crypto.h | 1 -
 3 files changed, 11 deletions(-)

diff --git a/crypto/api.c b/crypto/api.c
index 4308b1c8ca2e..2d90b95891c4 100644
--- a/crypto/api.c
+++ b/crypto/api.c
@@ -409,7 +409,6 @@ struct crypto_tfm *__crypto_alloc_tfm(struct crypto_alg *alg, u32 type,
 		goto out_err;
 
 	tfm->__crt_alg = alg;
-	refcount_set(&tfm->refcnt, 1);
 
 	err = crypto_init_ops(tfm, type, mask);
 	if (err)
@@ -508,7 +507,6 @@ static void *crypto_alloc_tfmmem(struct crypto_alg *alg,
 	tfm = (struct crypto_tfm *)(mem + tfmsize);
 	tfm->__crt_alg = alg;
 	tfm->node = node;
-	refcount_set(&tfm->refcnt, 1);
 
 	return mem;
 }
@@ -663,8 +661,6 @@ void crypto_destroy_tfm(void *mem, struct crypto_tfm *tfm)
 	if (IS_ERR_OR_NULL(mem))
 		return;
 
-	if (!refcount_dec_and_test(&tfm->refcnt))
-		return;
 	alg = tfm->__crt_alg;
 
 	if (!tfm->exit && alg->cra_exit)
diff --git a/crypto/internal.h b/crypto/internal.h
index e81cd7594b35..df3125897619 100644
--- a/crypto/internal.h
+++ b/crypto/internal.h
@@ -10,7 +10,6 @@
 
 #include <crypto/algapi.h>
 #include <linux/completion.h>
-#include <linux/err.h>
 #include <linux/jump_label.h>
 #include <linux/list.h>
 #include <linux/module.h>
@@ -169,10 +168,5 @@ static inline int crypto_is_test_larval(struct crypto_larval *larval)
 	return larval->alg.cra_driver_name[0];
 }
 
-static inline struct crypto_tfm *crypto_tfm_get(struct crypto_tfm *tfm)
-{
-	return refcount_inc_not_zero(&tfm->refcnt) ? tfm : ERR_PTR(-EOVERFLOW);
-}
-
 #endif	/* _CRYPTO_INTERNAL_H */
 
diff --git a/include/linux/crypto.h b/include/linux/crypto.h
index d354a2a7ac5f..e3c4be29aacc 100644
--- a/include/linux/crypto.h
+++ b/include/linux/crypto.h
@@ -642,7 +642,6 @@ int crypto_has_alg(const char *name, u32 type, u32 mask);
  */
 
 struct crypto_tfm {
-	refcount_t refcnt;
 
 	u32 crt_flags;
 
-- 
2.39.4

